### PR TITLE
Restructure buildCWMetric logic

### PR DESCRIPTION
**Pull request by kohrapha**
This PR restructures the code for building CW Metrics from `pdata.Metric` to remove duplicated logic. To do this, we had to define an interface `DataPoints` that wraps `pdata.(Int|Double|DoubleHistogram)DataPointSlice` and an interface `DataPoint` that wraps `pdata.(Int|Double|DoubleHistogram)DataPoint`. 

However, since the `At(i)` method for `DataPoints` requires returning a `DataPoint` which is not the same as what `pdata.(Int|Double|DoubleHistogram)DataPoint` returns, we have to define a light conversion struct that provides the correct method signature.

**Performance**
I ran the benchmarks implemented in #5 and compared the performance of this new code change against the old code, and got the following results:

```
benchmark                                                old ns/op     new ns/op     delta
BenchmarkTranslateOtToCWMetricWithInstrLibrary-12        9588          9182          -4.23%
BenchmarkTranslateOtToCWMetricWithoutInstrLibrary-12     8836          8475          -4.09%
BenchmarkTranslateOtToCWMetricWithNamespace-12           17.4          16.5          -5.17%
BenchmarkTranslateCWMetricToEMF-12                       23867         22283         -6.64%

benchmark                                                old allocs     new allocs     delta
BenchmarkTranslateOtToCWMetricWithInstrLibrary-12        131            122            -6.87%
BenchmarkTranslateOtToCWMetricWithoutInstrLibrary-12     116            112            -3.45%
BenchmarkTranslateOtToCWMetricWithNamespace-12           0              0              +0.00%
BenchmarkTranslateCWMetricToEMF-12                       45             45             +0.00%

benchmark                                                old bytes     new bytes     delta
BenchmarkTranslateOtToCWMetricWithInstrLibrary-12        7462          7190          -3.65%
BenchmarkTranslateOtToCWMetricWithoutInstrLibrary-12     6853          6869          +0.23%
BenchmarkTranslateOtToCWMetricWithNamespace-12           0             0             +0.00%
BenchmarkTranslateCWMetricToEMF-12                       84394         84391         -0.00%
```

We note that not only is there no negative impact on the performance, we see a minor improvement across the board. This is great news because, not only did we improve code readability for developers, performance was also not impacted.

### CODE REVIEW

**Comment by kohrapha** [comment]
Adding @mxiamxia, @hdj630, @shaochengwang as reviewers.

File: exporter/awsemfexporter/metric_translator.go [hunk 1]
```diff
@@ -75,9 +75,50 @@ type CWMetricStats struct {
...
+// Define wrapper interfaces such that At(i) returns a `DataPoint`
+type IntDataPointSlice struct {
+	pdata.IntDataPointSlice
+}
+type DoubleDataPointSlice struct {
+	pdata.DoubleDataPointSlice
+}
+type DoubleHistogramDataPointSlice struct {
+	pdata.DoubleHistogramDataPointSlice
+}
```
Comment (mxiamxia) [inline exporter/awsemfexporter/metric_translator.go:107]: missed `IntHistogramDataPointSlice`?

Comment (mxiamxia) [reply exporter/awsemfexporter/metric_translator.go:107]: oh. I see. we're not using IntHistogramDataPointSlice currently.

**Comment by mxiamxia** [comment]
Overall, LGTM. Can you please help pull the latest 2 PRs we made recently from upstream and update it in your PR? If possible, could you pls help finish the missing test case for TestGetMesurement()?

File: exporter/awsemfexporter/metric_translator.go (hunk 1 above)
Comment (kohrapha) [reply exporter/awsemfexporter/metric_translator.go:107]: yup!

**Comment by kohrapha** [comment]
@mxiamxia Merged upstream code and implemented test cases as mentioned.

**Review by mxiamxia — APPROVED** [review]

**Comment by mxiamxia** [comment]
Can you pls ping Shaocheng to see if he is also fine this PR? I think we can contribute this enhancement to upstream.

**Review by shaochengwang — APPROVED** [review]
LGTM!

**Pull request merged**